### PR TITLE
Add __repr__ methods

### LINKED ISSUE #482: Better `__repr__` for project models
Update `dipdup.models.Model` class to get better representation in crash reports. Now it's just `<Model>`.
Bonus points: do the same for other classes commonly discovered in crash reports.

### PR DESCRIPTION
**Pull request by fabrobles92**
Add repr to have better representation classes when reading logs and debugging

Closes #482

### CODE REVIEW

File: src/dipdup/context.py [hunk 1]
```diff
@@ -163,8 +163,8 @@ def __init__(
         self._handlers: dict[tuple[str, str], HandlerConfig] = {}
         self._hooks: dict[str, HookConfig] = {}
 
-    def __str__(self) -> str:
-        return pformat(self.__dict__)
```
Comment (droserasprout) [inline src/dipdup/context.py:167]: Please, remove the redundant `<Class` and ` >` here and in other places.

File: src/dipdup/package.py [hunk 2]
```diff
@@ -83,6 +83,9 @@ def __init__(self, root: Path, quiet: bool = False) -> None:
         self._types: dict[str, type[BaseModel]] = {}
         self._evm_abis = EvmAbiManager(self)
         self._cairo_abis = CairoAbiManager(self)
+    
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__} (root={self.root}, evm_abis={self._evm_abis}, cairo_abis={self._cairo_abis})>'
```
Comment (droserasprout) [inline src/dipdup/package.py]: Why have you chosen these two attributes specifically? 🤔 Please, leave only the `root` one.

File: src/dipdup/context.py [hunk 3]
```diff
@@ -745,6 +745,9 @@ def __init__(
         )
         self.logger = logger
         self.hook_config = hook_config
+    
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__}(logger={self.logger}, hook_config={self.hook_config})>'
```
Comment (droserasprout) [inline src/dipdup/context.py]: Please, leave only the hook name here.

File: src/dipdup/context.py [hunk 4]
```diff
@@ -809,6 +812,9 @@ def __init__(
             handler_config.parent.name if handler_config.parent else 'unknown',
             handler_config.parent._template_values if handler_config.parent else {},
         )
+    
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__}(handler_config={self.handler_config}, template_values={self.template_values})>'
```
Comment (droserasprout) [inline src/dipdup/context.py]: Please, leave only the handler name.

File: src/dipdup/dipdup.py [hunk 5]
```diff
@@ -612,6 +615,9 @@ def __init__(self, config: DipDupConfig) -> None:
         )
         self._index_dispatcher: IndexDispatcher = IndexDispatcher(self._ctx)
         self._schema: Schema | None = None
+    
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__}(config={self._config}>'
```
Comment (droserasprout) [inline src/dipdup/dipdup.py]: Leave only the package name here, please.

File: src/dipdup/index.py [hunk 6]
```diff
@@ -66,6 +66,9 @@ def __init__(
 
         self._logger = FormattedLogger(__name__, fmt=f'{config.name}: ' + '{}')
         self._state: models.Index | None = None
+    
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__}(ctx={self._ctx})>'
```
Comment (droserasprout) [inline src/dipdup/index.py]: Replace with index name, please.

File: src/dipdup/indexes/starknet_events/index.py [hunk 7]
```diff
@@ -35,6 +35,9 @@ def __init__(
     ) -> None:
         super().__init__(ctx, config, datasources)
         self._event_identifiers: dict[str, dict[str, str]] | None = None
+    
+    def __repr__(self) -> str:
```
Comment (droserasprout) [inline src/dipdup/indexes/starknet_events/index.py]: No need to redefine method from `Index` class.

File: src/dipdup/models/__init__.py [hunk 8]
```diff
@@ -394,6 +394,10 @@ class Model(TortoiseModel):
     def __init__(self, **kwargs: Any) -> None:
         super().__init__(**kwargs)
         self._original_versioned_data = self.versioned_data
+    
+    def __repr__(self) -> str:
+        pk = getattr(self, 'pk', None)
+        return f'<Class {self.__class__.__name__}(pk={pk}, versioned_data={self.versioned_data})>'
```
Comment (droserasprout) [inline src/dipdup/models/__init__.py]: Please, unwrap the `self.versioned_data` to `key=value` parts.

File: src/dipdup/models/__init__.py [hunk 9]
```diff
@@ -536,6 +540,9 @@ def __init_subclass__(cls) -> None:
         cls._misses = 0
         cls._cache = LRU(cls._maxsize)
         super().__init_subclass__()
+    
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__}(hits={self._hits}, misses={self._misses}, maxsize={self._maxsize})>'
```
Comment (droserasprout) [inline src/dipdup/models/__init__.py]: There is a `metrics` section in the crash report with all information about caches. Please, remove this definition.

File: src/dipdup/models/__init__.py [hunk 10]
```diff
@@ -696,6 +703,9 @@ class ContractMetadata(Model):
     class Meta:
         table = 'dipdup_contract_metadata'
         unique_together = ('network', 'contract')
+    
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__}(network={self.network}, contract={self.contract}, metadata={self.metadata})>'
```
Comment (droserasprout) [inline src/dipdup/models/__init__.py]: No need to redefine method from the parent class `Model`.

File: src/dipdup/models/__init__.py [hunk 11]
```diff
@@ -711,3 +721,6 @@ class TokenMetadata(Model):
     class Meta:
         table = 'dipdup_token_metadata'
         unique_together = ('network', 'contract', 'token_id')
+
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__}(network={self.network}, contract={self.contract}, token_id={self.token_id}, metadata={self.metadata})>'
```
Comment (droserasprout) [inline src/dipdup/models/__init__.py]: No need to redefine method from the parent class `Model`.

File: src/dipdup/transactions.py [hunk 12]
```diff
@@ -21,6 +22,9 @@ def __init__(
         self._immune_tables = immune_tables or set()
         self._transaction: dipdup.models.VersionedTransaction | None = None
         self._pending_updates: deque[dipdup.models.ModelUpdate] = deque()
+    
+    def __repr__(self) -> str:
+        return f'<Class {self.__class__.__name__} {pformat(self.__dict__)}>'
```
Comment (droserasprout) [inline src/dipdup/transactions.py]: Please, remove this definition, there's no useful information.

**Review by droserasprout — CHANGES_REQUESTED** [review]

File: src/dipdup/context.py (hunk 3 above)
Comment (fabrobles92) [reply src/dipdup/context.py:750]: @droserasprout To get the hook name would be self.template_values._index?

File: src/dipdup/context.py (hunk 4 above)
Comment (fabrobles92) [reply src/dipdup/context.py]: @droserasprout To get the handler name would be self.template_values._index or just handler_config.parent.name?

File: src/dipdup/context.py (hunk 3 above)
Comment (fabrobles92) [reply src/dipdup/context.py]: @droserasprout To get the hook name here how can I recover it from the self.hook_config? self.hook_config.name?

File: src/dipdup/context.py (hunk 4 above)
Comment (droserasprout) [reply src/dipdup/context.py]: I think the best format would be `<index>.<handler>`.

File: src/dipdup/context.py (hunk 3 above)
Comment (droserasprout) [reply src/dipdup/context.py]: Yes.

**Review by droserasprout — APPROVED** [review]

**Pull request merged**